ambry-messageformat/src/main/java/com.github.ambry.messageformat/BlobStoreRecovery.java

```diff
@@ -82,11 +82,14 @@
                           properties.getTimeToLiveInSeconds()));
               messageRecovered.add(info);
             } else {
-              boolean deleteFlag = MessageFormatRecord.deserializeDeleteRecord(stream);
-              MessageInfo info =
-                  new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(),
-                      deleteFlag);
-              messageRecovered.add(info);
+              DeleteRecord deleteRecord = MessageFormatRecord.deserializeDeleteRecord(stream);
+              if (deleteRecord.getVersion() == MessageFormatRecord.Delete_Version_V1) {
```

Comment:
can you explain why the `if` is required? Wouldn't delete record have some kind of values for account id, container id and operation time?